### PR TITLE
Create pyproject.toml

### PR DESCRIPTION
**Pull request by robertodr**
Adds requirement on `jupyter_packaging`

### CODE REVIEW

**Comment by hainm** [comment]
Thanks @robertodr but I want to keep it as is now. Python packaging is weird and I've spent many hours to debug the build/install issues (e.g: missing js files). My point is: we only need to update the code if it's broken. Do users need to have `jupyter_packaging`? If not, we can ignore that.

**Comment by hainm** [comment]
> Thanks @robertodr but I want to keep it as is now. Python packaging is weird and I've spent many hours to debug the build/install issues (e.g: missing js files).

e.g: This PR introduces new error `ModuleNotFoundError: No module named 'versioneer'`

**Comment by robertodr** [comment]
Agreed that only what's broken should be fixed, but indeed `setup.py` does `import jupyter_packaging`. When creating a binder instance, installation from PyPI will fail unless one explicitly lists `jupyter_packaging` as dependency. Would adding it to `install_requires` suffice to fix the problem?

**Comment by hainm** [comment]
I see your point. um.

```
$ pip install nglview --no-cache-dir
Collecting nglview
  Downloading nglview-3.0.0.tar.gz (5.7 MB)
     |████████████████████████████████| 5.7 MB 2.0 MB/s 
    ERROR: Command errored out with exit status 1:
....
        from jupyter_packaging import (
    ModuleNotFoundError: No module named 'jupyter_packaging'
    ----------------------------------------
...
  Downloading nglview-2.7.7.tar.gz (5.2 MB)

```

**Comment by hainm** [comment]
I am thinking about not using `jupyter_packaging`.

**Comment by hainm** [comment]
> Would adding it to install_requires suffice to fix the problem?

I am not sure if this will work because `jupyter_packaging` is imported in the top of the setup.py file.

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[build-system]
+requires = ["jupyter_packaging~=0.7.9", "setuptools>=40.8.0", "wheel"]
```
Comment (hainm) [inline pyproject.toml:2]: add `"versioneer-518"` too

Comment (hainm) [reply pyproject.toml:2]: https://github.com/python-versioneer/python-versioneer/issues/193#issuecomment-604025248

**Comment by hainm** [comment]
@robertodr I don't have better approach than your PR here, so let's go with it.

**Comment by hainm** [comment]
Thank you.

**Pull request merged**